spec.emu

```diff
@@ -130,6 +131,135 @@ location: https://github.com/tc39/proposal-decimal/
   <emu-clause id="sec-decimal-abstract-ops">
     <h1>Abstract Operations</h1>
 
+    <emu-clause id="sec-decimal128-roundtodecimal128domain" type="abstract operation">
+      <h1>RoundToDecimal128Domain(
+        _v_: a non-zero mathematical value,
+        optional _mode_: a Decimal128 rounding mode
+      ): *+∞*<sub>𝔻</sub>, *-∞*<sub>𝔻</sub>, *+0*<sub>𝔻</sub>, *-0*<sub>𝔻</sub>, or a mathematical value
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It computes the closest approximation to a given mathematical value, rounded according to the given rounding mode, that is available within the limits of Decimal128.</dd>
+      </dl>
+      <emu-alg>
+        1. If _mode_ is *undefined*, set _mode_ to *"halfEven"*.
+        1. Let _sign_ be 1.
+        1. If _v_ < 0, set _sign_ to -1.
+        1. If abs(_v_) < 1, then
+          1. Let _k_ be 0.
+          1. Let _approximation_ be 0.
+        1. Otherwise:
+          1. Let _k_ be the unique integer for which 10<sup>_k_</sup> ≤ abs(_v_) < 10<sup>_k_ + 1</sup>.
+          1. Let _approximation_ be _sign_ × 10<sup>_k_</sup>.
+        1. If _k_ > 6111, then
+          1. If _v_ < 0, return *-∞*<sub>𝔻</sub>.
+          1. Otherwise, return *+∞*<sub>𝔻</sub>.
+        1. Let _delta_ be _v_ - (_sign_ × _approximation_).
+        1. Let _n_ be the largest non-negative integer such that _n_ × 10<sup>_k_</sup> ≤ abs(_v_).
+        1. Assert: _n_ < 10.
+        1. Let _i_ be 0.
+        1. If _v_ > 1, then
+          1. While _i_ ≤ 33 and _delta_ > 1, perform the following steps:
+            1. Set _n_ to the largest non-negative integer such that _n_ < 10 and _n_ × 10<sup>_k_ + _i_</sup> ≤ abs(_delta_).
+            1. Set _approximation_ to _approximation_ + (_sign_ × _n_ × 10<sup>_k_ + _i_</sup>).
+            1. Set _delta_ to _v_ - (_sign_ × _approximation_).
+            1. Set _i_ to _i_ + 1.
+        1. If _delta_ > 1, then
+          1. If _v_ < 0, return *-∞*<sub>𝔻</sub>.
+          1. Otherwise, return *+∞*<sub>𝔻</sub>.
+        1. If _delta_ = 0, return _approximation_.
```

Comment:
Here _approximation_ is guaranteed to not be 0, right (otherwise delta would be non-zero)? Could you add an assertion to make it clearer?

Reply:
Right -- done.